test/ectest.c

```diff
@@ -2420,6 +2421,285 @@ static int custom_generator_test(int id)
     return ret;
 }
 
+/*
+ * check creation of curves from explicit params through the public API
+ */
+static int custom_params_test(int id)
+{
+    int ret = 0, nid, bsize;
+    const char *curve_name = NULL;
+    EC_GROUP *group = NULL, *altgroup = NULL;
+    EC_POINT *G2 = NULL, *Q1 = NULL, *Q2 = NULL;
+    const EC_POINT *P = NULL;
```

Comment:
It seems odd to use capitalised initial letters in these variable names. That goes against our normal style.

Reply:
`p` is declared below.  Having `P` and `p` seems like a recipe for confusion.

Reply:
> It seems odd to use capitalised initial letters in these variable names. That goes against our normal style.

@mattcaswell We already use capital letters for points in this file all over, and it is quite common in math.

I would argue that people looking at these tests working on the EC math will have been looking at articles and books where it is quite normal to use uppercase for points and lowercase for scalars.


> `p` is declared below. Having `P` and `p` seems like a recipe for confusion.

@paulidale true, it might be a cause for confusion: I will use a different capital letter, unless you prefer for this test to not use capital letters for points.

Reply:
See b3dfef102b400c906c9d171f29ff7f4f2da7fd3f

Reply:
See b3dfef102b400c906c9d171f29ff7f4f2da7fd3f